cmd/neutron_query_relayer/cmd/start.go

```diff
@@ -65,29 +70,40 @@ func startRelayer() {
 		logger.Fatal("cannot initialize relayer config", zap.Error(err))
 	}
 
-	http.Handle("/metrics", promhttp.Handler())
-	go func() {
-		err := http.ListenAndServe(fmt.Sprintf(":%d", cfg.PrometheusPort), nil)
-		if err != nil {
-			logger.Fatal("failed to serve metrics", zap.Error(err))
-		}
-	}()
-	logger.Info("metrics handler set up")
-
 	ctx, cancel := context.WithCancel(context.Background())
 	wg := &sync.WaitGroup{}
 
 	// The storage has to be shared because of the LevelDB single process restriction.
 	storage, err := app.NewDefaultStorage(cfg, logger)
 	if err != nil {
-		logger.Fatal("Failed to create NewDefaultStorage", zap.Error(err))
+		logger.Fatal("failed to create NewDefaultStorage", zap.Error(err))
 	}
 	defer func(storage relay.Storage) {
 		if err := storage.Close(); err != nil {
-			logger.Error("Failed to close storage", zap.Error(err))
+			logger.Error("failed to close storage", zap.Error(err))
 		}
 	}(storage)
 
+	http.Handle("/metrics", monitoring.NewPromWrapper(logRegistry, storage))
```

Comment:
I see one point for it staying as it were:
It's just hard separates these two servers so they are independent.
Why:
- We can configure metrics and webserver access in different way (because different ports)
- Some kind of trouble in one thing does not bring down another (debatable, but still)

Reply:
> We can configure metrics and webserver access in different way (because different ports)

we can, but why? isn't different endpoints enough?

> Some kind of trouble in one thing does not bring down another (debatable, but still)

what can go wrong with a web server? it just handles requests. if a request goes bad, only this request's scope is affected, the web server execution persists

Reply:
agreed, other things can be resolved with reverse-proxy